src/modules/projects/components/ExternalSubmissionsViewModal.tsx

```diff
@@ -143,12 +142,10 @@ const ExternalSubmissionsModalContent = ({
             </Alert>
           )}
           <Card sx={{ p: 2 }}>
-            {submission && (
-              <DynamicView
-                values={submissionValues}
-                definition={submission.definition.definition}
-              />
-            )}
+            <DynamicView
```

Comment:
Remove guards in this component against falsy `submission`, since with the code restructure it will only be rendered when `submission` is present.